x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/action/RenderMetadataAction.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.search.action;
+
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+public class RenderMetadataAction extends ActionType<RenderMetadataAction.Response> {
+
+    public static final RenderMetadataAction INSTANCE = new RenderMetadataAction();
+    public static final String NAME = "cluster:admin/xpack/application/search_application/render_metadata";
+
+    public RenderMetadataAction() {
+        super(NAME, RenderMetadataAction.Response::new);
+    }
+
+    public static class Response extends ActionResponse implements ToXContentObject {
+
+        private final Map<String, Object> renderedTemplateParams;
+
+        public Response(StreamInput in) throws IOException {
+            super(in);
+            this.renderedTemplateParams = in.readMap();
+        }
+
+        public Response(Map<String, Object> renderedTemplateParams) {
+            this.renderedTemplateParams = renderedTemplateParams;
```

Comment:
I think we need to check for non-null params
```suggestion
            Objects.requireNonNull(renderedTemplateParams, "renderedTemplateParams must not be null");
            this.renderedTemplateParams = renderedTemplateParams;
```

Reply:
Good catch, thanks!